storage/kubernetes/controller.go

```diff
@@ -160,7 +160,16 @@ func (s *storage) targetSecret() (*v1.Secret, error) {
 }
 
 func (s *storage) saveInK8s(secret *v1.Secret) (*v1.Secret, error) {
-	if !s.initComplete() {
+
+	// wait for storage init
+	err := wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, 1*time.Minute, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
1. Polling with a context, but then using `context.Background()` doesn't make a lot of sense
2. This makes the `saveInk8s` function block for up to a minute when updating secrets if the apiserver isn't available, which is unacceptable for K3s. dynamiclistener is started very early, and the apiserver later wired into its request handlers. Dynamiclistener needs to fail fast and write the secret through to the backing storage immediately, if the secret controller isn't up yet.

Reply:
Issues with use of context aside, It sounds like we maybe need to make this configurable so that different projects can have their desired behavior. 

If the apiserver is unavalable:
* K3s wants the backing storage updated immediately, and the Kubernetes secret synced later once the apiserver is up.
* Rancher wants to block on the apiserver coming up, and not update the backing storage until after the secret has been updated.